backend/transcribee_backend/routers/task.py

```diff
@@ -64,67 +78,101 @@ def claim_unassigned_task(
         .where(
             col(Task.task_type).in_(task_type),
             or_(
-                is_(Task.assigned_worker_id, None),
-                col(Task.last_keepalive)
-                < now_tz_aware() - datetime.timedelta(seconds=settings.worker_timeout),
+                is_(Task.current_attempt_id, None),
+                Task.current_attempt.has(
+                    TaskAttempt.last_keepalive
+                    < now_tz_aware()
+                    - datetime.timedelta(seconds=settings.worker_timeout)
+                ),
             ),
-            is_(Task.is_completed, False),
+            ~(col(Task.state).in_([TaskState.COMPLETED, TaskState.FAILED])),
             ~blocking_tasks_exist,
         )
         .with_for_update()
     )
-    task = session.exec(statement).first()
-    if task is None:
-        return task
-    task.assigned_worker = authorized_worker
-    task.last_keepalive = now_tz_aware()
-    task.assigned_at = now_tz_aware()
+    return session.exec(statement).first()
+
+
+@task_router.post("/claim_unassigned_task/")
+def claim_unassigned_task(
+    session: Session = Depends(get_session),
+    authorized_worker: Worker = Depends(get_authorized_worker),
+    task_type: List[TaskType] = Query(),
+    now: datetime.datetime = Depends(now_tz_aware),
+) -> Optional[AssignedTaskResponse]:
+    while True:
+        task = get_ready_task(session, task_type)
+        if task is None:
+            session.commit()  # We might have marked tasks as failed
+            return
+        elif task.remaining_attempts <= 0:
+            mark_task_as_failed(session, task, now)
```

Comment:
A bit sad is, that we will not mark a task as failed if the task crashes the only connected worker and it does not come back?